helix-core/src/match_brackets.rs

```diff
@@ -9,16 +9,32 @@ use crate::Syntax;
 const MAX_PLAINTEXT_SCAN: usize = 10000;
 const MATCH_LIMIT: usize = 16;
 
-// Limit matching pairs to only ( ) { } [ ] < > ' ' " "
-const PAIRS: &[(char, char)] = &[
+pub const BRACKETS: [(char, char); 7] = [
     ('(', ')'),
     ('{', '}'),
     ('[', ']'),
     ('<', '>'),
-    ('\'', '\''),
-    ('\"', '\"'),
+    ('«', '»'),
+    ('「', '」'),
+    ('（', '）'),
 ];
 
+// The difference between BRACKETS and PAIRS is that we can find matching
+// BRACKETS in a plain text file, but we can't do the same for PAIRs.
+// PAIRS also contains all BRACKETS.
+pub const PAIRS: [(char, char); BRACKETS.len() + 3] = {
+    let mut pairs = [(' ', ' '); BRACKETS.len() + 3];
+    let mut idx = 0;
+    while idx < BRACKETS.len() {
+        pairs[idx] = BRACKETS[idx];
+        idx += 1;
+    }
+    pairs[idx] = ('"', '"');
+    pairs[idx + 1] = ('\'', '\'');
+    pairs[idx + 2] = ('`', '`');
+    pairs
+};
```

Comment:
hopefully more std functions become const so we can use copy_from_slice here for exampple